test/unit/node_stream_spec.js

```diff
@@ -0,0 +1,191 @@
+/* Copyright 2017 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* globals __non_webpack_require__ */
+
+import { isNodeJS } from '../../src/shared/util';
+import { PDFNodeStream } from '../../src/display/node_stream';
+
+describe('node_stream', function() {
+  let pdf1 =
+    'http://mozilla.github.io/pdf.js/web/compressed.tracemonkey-pldi-09.pdf';
```

Comment:
This results in an external network request, and slow down the unit test if network is slow, or even cause the unit test to fail if network is somehow unavailable.

Please use a local file server in Node for testing. Don't forget to close the server regardless of test failure. To do so you can use `http.createServer in a `beforeAll` hook (and start the server on a random free port using `.listen(0)`, and call `.close()` on the server in an `afterAll` hook. The return value of `.listen` has an `address` method that contains the random port in the `port` property. After starting the server, you can use something like `http://127.0.0.1:${port}/tracemonkey.pdf`.


And for the browser... Why does this test have a special case `!isNode()` if the API that you are testing is only supposed to work on Node? Am I missing something?